pennylane_lightning/lightning_qubit/lightning_qubit2.py

```diff
@@ -0,0 +1,509 @@
+            TensorLike, tuple[TensorLike], tuple[tuple[TensorLike]]: A numeric result of the computation.
+        """
+        results = []
+        for circuit in circuits:
+            circuit = circuit.map_to_standard_wires()
+            results.append(simulate(circuit, self._statevector))
+
+        return tuple(results)
+
+    def supports_derivatives(
+        self,
+        execution_config: Optional[ExecutionConfig] = None,
+        circuit: Optional[qml.tape.QuantumTape] = None,
+    ) -> bool:
+        """Check whether or not derivatives are available for a given configuration and circuit.
+
+        ``LightningQubit2`` supports adjoint differentiation with analytic results.
+
+        Args:
+            execution_config (ExecutionConfig): The configuration of the desired derivative calculation
+            circuit (QuantumTape): An optional circuit to check derivatives support for.
+
+        Returns:
+            Bool: Whether or not a derivative can be calculated provided the given information
+
+        """
+        if execution_config is None and circuit is None:
+            return True
+        if execution_config.gradient_method not in {"adjoint", "best"}:
+            return False
+        if circuit is None:
+            return True
+        return _supports_adjoint(circuit=circuit)
+
+    def compute_derivatives(
+        self,
+        circuits: QuantumTape_or_Batch,
+        execution_config: ExecutionConfig = DefaultExecutionConfig,
+    ):
+        """Calculate the jacobian of either a single or a batch of circuits on the device.
+
+        Args:
+            circuits (Union[QuantumTape, Sequence[QuantumTape]]): the circuits to calculate derivatives for
+            execution_config (ExecutionConfig): a datastructure with all additional information required for execution
+
+        Returns:
+            Tuple: The jacobian for each trainable parameter
+        """
+        batch_obs = execution_config.device_options.get("batch_obs", self._batch_obs)
+        return tuple(
+            jacobian(circuit, self._statevector, batch_obs=batch_obs) for circuit in circuits
+        )
+
+    def execute_and_compute_derivatives(
+        self,
+        circuits: QuantumTape_or_Batch,
+        execution_config: ExecutionConfig = DefaultExecutionConfig,
+    ):
+        """Compute the results and jacobians of circuits at the same time.
+
+        Args:
+            circuits (Union[QuantumTape, Sequence[QuantumTape]]): the circuits or batch of circuits
+            execution_config (ExecutionConfig): a datastructure with all additional information required for execution
+
+        Returns:
+            tuple: A numeric result of the computation and the gradient.
+        """
+        batch_obs = execution_config.device_options.get("batch_obs", self._batch_obs)
+        results = tuple(
+            simulate_and_jacobian(c, self._statevector, batch_obs=batch_obs) for c in circuits
+        )
+        return tuple(zip(*results))
```

Comment:
Should we check for gradient_method here? In case someone asks for a not-supported gradient method what would be a reasonable return from this method?

Reply:
This is a preprocessing step handled by `dev.preprocess`, which is called in `qml.execute`. It's stated in the documentation that using `dev.execute/compute_derivatives` is not guaranteed to work with tapes that are not preprocessed.